processor/error/payload.go

```diff
@@ -18,9 +19,40 @@ type payload struct {
 	Service m.Service
 	System  *m.System
 	Process *m.Process
-	User    common.MapStr
+	User    *m.User
 
-	Events []Event `mapstructure:"errors"`
+	Events []Event
+}
+
+func decodeError(raw map[string]interface{}) (*payload, error) {
+	if raw == nil {
+		return nil, nil
+	}
+	pa := &payload{}
+	var err error
+	service, err := m.DecodeService(raw["service"], err)
+	if service != nil {
+		pa.Service = *service
+	}
+	pa.System, err = m.DecodeSystem(raw["system"], err)
+	pa.Process, err = m.DecodeProcess(raw["process"], err)
+	pa.User, err = m.DecodeUser(raw["user"], err)
+	if err != nil {
+		return nil, err
+	}
+
+	df := utility.DataFetcher{Err: err}
```

Comment:
if `err != nil`, you are returning in L41, so here Err: is going to be always nil, so you don'y need to pass it.